ncm-openstack/src/main/pan/components/openstack/common.pan

```diff
@@ -0,0 +1,155 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+
+
+declaration template components/openstack/common;
+
+type type_storagebackend = string with match(SELF, '^(file|http|swift|rbd|sheepdog|cinder|vmware)$');
+
+type type_neutrondriver = string with match(SELF, '^(local|flat|vlan|gre|vxlan|geneve)$');
+
+type type_neutronextension = string with match(SELF, '^(qos|port_security)$');
+
+type type_directory = string with match(SELF, '[^/]+/?$');
```

Comment:
there's a type for that already: https://github.com/quattor/template-library-core/blob/571476a475924ef1ad0f0bd8f65241d2adc48d3f/pan/types.pan#L976